### PR TITLE
silkswap on bahamut chain

### PR DESCRIPTION
**Pull request by madi-blockstars**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
SilkSwap

##### Twitter Link:
@Blockstarstech

##### List of audit links if any:
https://r.xyz/program/silkswap

##### Website Link:
https://silkswap.me/

##### Chain:
Bahamut Mainnet
https://www.ftnscan.com/

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)

https://www.coingecko.com/en/exchanges/silkswap

##### Short Description (to be shown on DefiLlama):

SilkSwap - first DEX on Bahamut! Enjoy silk smooth swapping with us,  explore diverse assets, trade peer-to-peer, and retain control over your funds securely. Enjoy fast, low-cost transactions, access real-time data, and make informed trades. Join Silk Swap for seamless, secure, permissionless and efficient trading.

##### Token address and ticker if any:
this is smart contract address
https://www.ftnscan.com/address/0xF660558a4757Fb5953d269FF32E228Ae3d5f6c68/

##### Category (full list at https://defillama.com/categories) *Please choose only one:

4.Dexes

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
N/A
##### Implementation Details: Briefly describe how the oracle is integrated into your project:

smart contract is a clone of uniswap V2

##### forkedFrom (Does your project originate from another project):

smart contract is a fork of uniswap V2

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/blockstars-tech/SilkSwapDEXSmartContracts

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/silkswap: 
        
 
  
    
    
    Error: 
        Unknown chain(s): bahamut
        Note: if you think that the chain is correct but missing from our list, please add it to 'projects/helper/chains.json' file
        
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:152:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

File: projects/silkswap/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const { getChainTvl } = require('../helper/getUniSubgraphTvl');
+const { getUniTVL } = require('../helper/unknownTokens');
+
+const v2graph = getChainTvl({
+  bahamut: "https://bahamut-rpc.publicnode.com"
```
Comment (Define101) [inline projects/silkswap/index.js:5]: ftn instead of bahamut

File: projects/silkswap/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+})
+
+//I could not find bahamut chain in this repo even tho it is listed on website 
+// https://defillama.com/chain/Bahamut
+//and swap address on bahamut https://www.ftnscan.com/address/0xF660558a4757Fb5953d269FF32E228Ae3d5f6c68/
+module.exports = {
+  misrepresentedTokens: true,
+  methodology: `its a dex which operates on bahamut chain `,
+  bahamut: {
```
Comment (Define101) [inline projects/silkswap/index.js:14]: same here , ftn

File: projects/silkswap/index.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    tvl: v2graph('bahamut'),
+  },
+}
+
+const config = {
+"bahamut" : "0xF660558a4757Fb5953d269FF32E228Ae3d5f6c68" //address of smart contract on bahamut chain
```
Comment (Define101) [inline projects/silkswap/index.js:20]: same here, ftn

**Comment by llamatester** [comment]
Error while running adapter at projects/silkswap: 
        
 
  
    
    
    Error: Llama RPC error! methos: undefined 
    - host: https://rpc1.bahamut.io error: execution reverted
    - host: https://bahamut-rpc.publicnode.com error: execution reverted
    - host: https://rpc2.bahamut.io error: execution reverted

**Comment by g1nt0ki** [comment]
@madi-blockstars thanks for the PR

**Comment by llamatester** [comment]
The adapter at projects/silkswap exports TVL: 
        
 
 
    ftn                       127.26 k
    
    total                    127.26 k

**Pull request merged**